ddtrace/tracer/textmap.go

```diff
@@ -386,6 +386,9 @@ func (p *propagator) extractTextMap(reader TextMapReader) (ddtrace.SpanContext,
 			ctx.origin = v
 		case traceTagsHeader:
 			unmarshalPropagatingTags(&ctx, v)
+			if len(ctx.trace.propagatingTag(keyTraceID128)) != 16 {
+				ctx.trace.unsetPropagatingTag(keyTraceID128)
+			}
```

Comment:
Can we do this below where we're doing the hex parsing instead? Checking the validity of `keyTraceID128` in two places makes this a bit more complicated than it needs to be. Plus, we also want to unset this propagating tag if the hex is invalid, so the logic should be the same. Right now, we're still keeping the `_dd.p.tid` tag even if it's invalid hex, which isn't correct per the RFC:
> if the tag value is a valid 32 character lower-case hexadecimal encoded number, set the higher-order 64 bits of the TraceId to its (decoded) value.
**if the tag value is malformed, discard the tag value** and, optionally, set the `dd.propagation_error` tag to:
`malformed_tid ` + <_dd.p.tid value>

Reply:
If the key is a valid 15-char string, it isn't a valid tid, and thus should be discarded.